application/src/main/java/org/togetherjava/tjbot/features/mediaonly/MediaOnlyChannelListener.java

```diff
@@ -45,19 +54,60 @@ public void onMessageReceived(MessageReceivedEvent event) {
             return;
         }
 
-        if (messageHasNoMediaAttached(message)) {
-            message.delete().flatMap(any -> dmUser(message)).queue(any -> {
-            }, failure -> tempNotifyUserInChannel(message));
+        long userId = event.getAuthor().getIdLong();
+
+        boolean isForwardedWithMedia =
+                !message.getMessageSnapshots().isEmpty() && !messageHasNoMediaAttached(message);
+
+        if (isForwardedWithMedia) {
+            lastValidForwardedMediaMessageTime.put(userId, System.currentTimeMillis());
+            return;
+        }
+
+        boolean isNormalMediaUpload =
+                message.getMessageSnapshots().isEmpty() && !messageHasNoMediaAttached(message);
+        if (isNormalMediaUpload) {
+            return;
+        }
+
+        Long lastForwardedMediaTime = lastValidForwardedMediaMessageTime.get(userId);
+        long gracePeriodMillis = TimeUnit.SECONDS.toMillis(1);
+
+        if (lastForwardedMediaTime != null
+                && (System.currentTimeMillis() - lastForwardedMediaTime) <= gracePeriodMillis) {
+            lastValidForwardedMediaMessageTime.remove(userId);
+            return;
         }
+
+        message.delete().queue(deleteSuccess -> dmUser(message).queue(dmSuccess -> {
+        }, dmFailure -> tempNotifyUserInChannel(message)),
+                deleteFailure -> tempNotifyUserInChannel(message));
     }
 
     private boolean messageHasNoMediaAttached(Message message) {
-        return message.getAttachments().isEmpty() && message.getEmbeds().isEmpty()
-                && !message.getContentRaw().contains("http");
+        if (!message.getAttachments().isEmpty() || !message.getEmbeds().isEmpty()
+                || MEDIA_URL_PATTERN.matcher(message.getContentRaw()).matches()) {
+            return false;
+        }
+
+        if (!message.getMessageSnapshots().isEmpty()) {
+            for (MessageSnapshot snapshot : message.getMessageSnapshots()) {
+                if (!snapshot.getAttachments().isEmpty() || !snapshot.getEmbeds().isEmpty()
+                        || MEDIA_URL_PATTERN.matcher(snapshot.getContentRaw()).matches()) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        return true;
     }
 
     private MessageCreateData createNotificationMessage(Message message) {
         String originalMessageContent = message.getContentRaw();
+        if (originalMessageContent.trim().isEmpty()) {
+            originalMessageContent = "(Original message had no visible text content)";
```

Comment:
that text is written with a developer as target audience, not the user. id rewrite it as `See image:` or similar.

Reply:
what i mean is that if this is a user facing text, its not written in a user friendly way.